### PR TITLE
Add Bepuphysics benchmarks

### PR DESCRIPTION
**Pull request by kunalspathak**
Adds benchmarks for physics engine written in .NET by [BEPU physics](https://www.bepuentertainment.com/).

This is a snapshot of `DemoBenchmarks` taken from https://github.com/bepu/bepuphysics2/commit/d9b1eea65b55f43e61b39de51f34db3a0d2cf5a1. 

@RossNordby gave permissions to include `bepuphysics` benchmarks in `dotnet/performance` repo. See https://github.com/bepu/bepuphysics2/issues/239

Need to run with the following to run the trimmed version as mentioned in https://github.com/bepu/bepuphysics2/issues/239#issuecomment-1298767905 and in `README.md`.

```
-f *CollisionBatcherTaskBenchmarks.* *GroupedCollisionTesterBenchmarks.* *GatherScatterBenchmarks.* *OneBodyConstraintBenchmarks.* *TwoBodyConstraintBenchmarks.* *ThreeBodyConstraintBenchmarks.* *FourBodyConstraintBenchmarks.* *SweepBenchmarks.* *ShapeRayBenchmarks.* *ShapePileBenchmark.* *RagdollTubeBenchmark.*
```

### CODE REVIEW

**Comment by kunalspathak** [comment]
With trimmed version, it just takes 12 minutes.

**Comment by kunalspathak** [comment]
@sblom @danmoseley

**Comment by sblom** [comment]
This is excellent! Perf team will make sure we get these plumbed through.

**Comment by sblom** [comment]
@kunalspathak I'm having @caaavik-msft take a look at it in our current sprint. Expect him to follow up directly with any questions.

**Comment by kunalspathak** [comment]
> Perf team will make sure we get these plumbed through.

That will be great. We have other benchmarks in `real-world` folder that are not ran today. Hoping to get it in the pipeline where they are run daily and bugs reported for any regression.

File: NuGet.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (caaavik-msft) [inline NuGet.config:1]: Not sure why this file changed, did the line ending change maybe?

Comment (kunalspathak) [reply NuGet.config]: fixed.

**Review by caaavik-msft — APPROVED** [review]

**Pull request merged**